### ISSUE TITLE
Add an extension to get all attributes on an NSAttributedString

### ISSUE THREAD

**Issue by SD10** [SwifterSwift/SwifterSwift#333]
Add an extension to get all attributes on an NSAttributedString:

```Swift
extension NSAttributedString {
    var attributes: [NSAttributedStringKey: Any] {
        /// implementation ...
    }
}
```

**Comment by nathanbacon** [comment]
Hey Steven, I'm running into trouble with this. There are two types of possible strings this could be applied to:

One where the attributed is applied across the whole string as in
`NSAttributedString("Test").bolded`

and another where there are multiple attributes of the same type across the string
`NSAttributedString("Red").colored(with: UIColor.red) + NSAttributedString("Blue").colored(with: UIColor.blue)`

The problem with the second type is that the returned dictionary could only contain one color because both portions have the same NSAtributedStringKey. Returning an Array<(NSAttributedStringKey, Any, NSRange)> would be better for this case.

If this extension is meant to work on homogenous strings, the solution is easy. Let me know what you think.

Thanks

**Comment by SD10** [comment]
Hey @nathanbacon!
I was only thinking of homogenous Strings so it is indeed an easy extension. If you think you have a solution for strings with multiple attributes under a single key, I’d look at that too

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
Closed by #340
